src/common/tls_session.cc

```diff
@@ -0,0 +1,83 @@
+/*
+   Copyright 2023 Leil Storage OÜ
+
+   This file is part of SaunaFS.
+
+   SaunaFS is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, version 3.
+
+   SaunaFS is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with SaunaFS  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "common/platform.h"
+
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include <openssl/x509v3.h>
+
+#include "common/tls_session.h"
+
+const std::string TlsSession::kNoFile = "/unused file/";
+
+TlsSession::TlsSession(int socket, bool isServer, const std::string &keyFile,
+                       const std::string &certFile, const std::string &trustedCAFile,
+                       const std::string expectedHostname) {
```

Comment:
consider a const reference here:

```suggestion
                       const std::string &expectedHostname) {
```

Reply:
Done.